[Leia] fix playback error and prevent password ask, if not enabled in settings

AlwinEsch opened the pull request:
This fixes some problems during playback of videos from RAR files.

Also are settings added to enable (disabled as default) user password ask (by Kodi dialog) and to give 5 standard passwords in settings where tried first.

Further is now during Kodi stop the stored password list checked and entries of no more present RAR packages removed.

**WARNING: About settings is a bad bug inside Kodi itself!!! If addon enabled can Kodi not store the settings, addon must be disabled and kodi restartet before to store them and then enabled and restarted again!**
See https://github.com/xbmc/xbmc/pull/17450 (Leia fix) and https://github.com/xbmc/xbmc/pull/17449 (Matrix fix)

AlwinEsch committed:
remove no more needed codecvt.h header

AlwinEsch committed:
increase version to 2.2.1

AlwinEsch committed:
fix direct file read on RAR files

Before was some errors inside and also was on encrypted RAR's the seek
not possible.

garbear commented:
Cool, can you post a screenshot of how the settings dialog looks now? I'll supply some grammar improvements.

AlwinEsch commented:
![Bildschirmfoto von 2020-03-09 11-06-43](https://user-images.githubusercontent.com/6879739/76203475-fc3e1e80-61f6-11ea-8f24-f201d7f355e4.png)
![Bildschirmfoto von 2020-03-09 11-11-20](https://user-images.githubusercontent.com/6879739/76203516-11b34880-61f7-11ea-8c9d-17a5c91255c3.png)

Here the screenshots how it is used, I think the text could really be better

AlwinEsch commented:
Because of the first if deactivated, the RARs are then also shown in the "Kodi Settings->File Manager", but cannot be opened (without a password), moreover comes with others where the archive itself is unencrypted, but the files in it are encrypted they also appear and shown (but not possible to open).

Unfortunately it is not yet possible to give a extra Value to ListItem (via addon) so that the Kodi shows that it is encrypted.

AlwinEsch committed:
fix wrong file size give, before it taken only his lower 32bit

RAR use two different values to give 64 bit (think as backward to older API).
There two values UnpSizeHigh and UnpSize must be compinated to have right.

AlwinEsch committed:
add setting to enable pw ask, disabled as default and allow default pw's

This is to prevent bad open of password ask on to much files.

Also alow this to store 5 passwords in settings to check against.
